common/arch/sdl/window.cpp

```diff
@@ -72,21 +92,19 @@ int window_close(window *wind)
 		return 0;
 	}
 
-	if (wind == FrontWindow)
-		FrontWindow = wind->prev;
-	if (wind == FirstWindow)
-		FirstWindow = wind->next;
-	if (wind->next)
-		wind->next->prev = wind->prev;
-	if (wind->prev)
-		wind->prev->next = wind->next;
+	if (type == window_alloc_type::separate)
+		wind->unlink();		// Remove from the window list
```

Comment:
Why do freestanding window instances need one form of unlink, but embedded instances get a different form?  Would it not be simpler to say that `window_close` will always unlink the window and that deletion must always be performed by a function which knows the most-derived type?

The current design looks like it would be easy for someone to do the wrong thing by allocating a window with the type set to `subclass`, then calling `window_close` on it and not explicitly deleting the allocated window.  This could be surprising, since `window_close` will implicitly delete some windows, but not others.